sympy/series/limits.py

```diff
@@ -146,15 +146,14 @@ def doit(self, **hints):
 
         if e.is_Mul:
             if abs(z0) is S.Infinity:
-                # XXX todo: this should probably be stated in the
-                # negative -- i.e. to exclude expressions that should
-                # not be handled this way but I'm not sure what that
-                # condition is; when ok is True it means that the leading
-                # term approach is going to succeed (hopefully)
+                ztemp = Symbol("z",real = True)
```

Comment:
@jksuom ,Well if we have function like `e**(x**2 + x + 2)` the log is calculated and the leading term rule is applied on polynomial. Log can only be calculated by a variable in real domain.So, I have defined this variable.